### PR TITLE
log backup: fix the issue that paused service safepoint not removed

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/52082

Problem Summary:
stop task didn't remove service safepoint correctly.

### What changed and how does it work?
1. set ttl == 0 to make pd service remove this safepoint.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that stop log backup didn't clean paused task gc safepoint correctly.
```

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -724,8 +725,8 @@ func RunStreamStop(
 	if err := streamMgr.setGCSafePoint(ctx,
 		utils.BRServiceSafePoint{
 			ID:       buildPauseSafePointName(ti.Info.Name),
-			TTL:      utils.DefaultStreamStartSafePointTTL,
-			BackupTS: 0,
+			TTL:      0, // 0 means remove this service safe point.
```
[the remark this answers is not in the document]
Comment (3pointer) [reply br/pkg/task/stream.go:728]: yes, and if set ts to 0. it won't pass the check before connect to pd to remove this safepoint.

**Review by Leavrth — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**